### ISSUE TITLE
Style setter does not support `Binding`

### ISSUE THREAD

**Issue by MartinZikmund** [unoplatform/uno#4826]
## Current behavior

When `Binding` is used in `Style` `Setter`, Uno targets don't compile.

## Expected behavior

Should compile.

`TabView` button style uses this approach, so it would be a useful addition.

## How to reproduce it (as minimally and precisely as possible)

```
<Page.Resources>
   <SolidColorBrush x:Key="TestBrush">Red</SolidColorBrush>
   <Style TargetType="TextBlock">
      <Setter Property="Foreground" Value="{Binding Source={ThemeResource TestBrush}}" />
   </Style>
</Page.Resources>
<Grid>
   <TextBlock Text="Test" />
</Grid>
```

Run the app on Windows and notice the text is red.

Try to compile under Uno and notice build fails with _error CS1024: Preprocessor directive expected_

The error is also in the generated `g.cs` file, similar to the following:

```
new global::Windows.UI.Xaml.Setter(global::Windows.UI.Xaml.Controls.TextBlock.ForegroundProperty, 
__ResourceOwner_278, __ResourceOwner_279 => (Windows.UI.Xaml.SolidColorBrush)
						#Error // Binding could not be found.
)
```
## Workaround

Avoid use of Binding in style setter

## Environment

Nuget Package:

- [x] Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia
- [ ] Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia
- [ ] Uno.SourceGenerationTasks
- [ ] Uno.UI.RemoteControl / Uno.WinUI.RemoteControl
- [ ] Other: 

Nuget Package Version(s):

Affected platform(s):

- [x] iOS
- [x] Android
- [x] WebAssembly
- [x] WebAssembly renderers for Xamarin.Forms
- [x] macOS
- [x] Skia
  - [x] WPF
  - [x] GTK (Linux)
  - [x] Tizen
- [ ] Windows
- [ ] Build tasks
- [ ] Solution Templates

IDE:

- [ ] Visual Studio 2017 (version: )
- [x] Visual Studio 2019 (version: )
- [ ] Visual Studio for Mac (version: )
- [ ] Rider Windows (version: )
- [ ] Rider macOS (version: )
- [ ] Visual Studio Code (version: )

Relevant plugins:

- [ ] Resharper (version: )

**Comment by MartinZikmund** [comment]
When fixed, revert Uno specific TODOs in both `TabView.xaml` files

**Comment by davidjohnoliver** [comment]
Strange that `TabView` is using this - I was under the impression it wasn't supported on UWP? https://stackoverflow.com/questions/33573929/uwp-binding-in-style-setter-not-working

**Comment by MartinZikmund** [comment]
@davidjohnoliver They are using it this way:

```xaml
<contract7Present:Setter Property="CornerRadius" Value="{Binding Source={ThemeResource OverlayCornerRadius}, Converter={StaticResource TopCornerRadiusFilterConverter}}"/>
```

So maybe this use-case is supported - to utilize `{Binding}` for the purposes of a value converter.

**Comment by davidjohnoliver** [comment]
Ahhhh...... yeah, maybe it works in the particular case that a Source is specified (and it's a resource)

**Comment by jeromelaban** [comment]
We'll probably have to make a special case for those. it's probably excluded explicitly in the generator for setters.

**Comment by davidjohnoliver** [comment]
This probably could be supported using `SetterExtensions.ApplyThemeResourceUpdateValues()`, similar to as is done when the value is directly set to a ThemeResource:

https://github.com/unoplatform/uno/blob/55101581bdd4b2faccab84d7a52fc1b548fc8cb5/src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs#L1592-L1595

**Comment by MartinZikmund** [comment]
Ensure the fix also supports the following scenario:

[SetterConvertedBug.zip](https://github.com/unoplatform/uno/files/8900676/SetterConvertedBug.zip)


Given VS 2019, create new UNO solution, leave only 3 projects: Shared, UWP and Wasm
Add some value converter:

```
    public class SomeConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, string language)
        {
            return (parameter ?? "").ToString();
        }

        public object ConvertBack(object value, Type targetType, object parameter, string language)
        {
            throw new NotImplementedException();
        }
    }
```
Use it in MainPage.xaml:

```
<Page
    x:Class="SetterConvertedBug.MainPage"
    xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:local="using:SetterConvertedBug"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
    mc:Ignorable="d"
    Background="{ThemeResource ApplicationPageBackgroundThemeBrush}">

    <Page.Resources>
        <local:SomeConverter x:Key="SomeConverter"/>
    </Page.Resources>

    <Grid>
        <TextBlock>
            <TextBlock.Resources>
                <Style TargetType="TextBlock">
                    <Setter Property="Text">
                        <Setter.Value>
                            <Binding Path="Whatever" Converter="{StaticResource SomeConverter}"/>
                        </Setter.Value>
                    </Setter>
                </Style>
            </TextBlock.Resources>
        </TextBlock>
    </Grid>
</Page>
```

Build.
UWP build is OK, Wasm build fails with error at Uno.UI.SourceGenerators.XamlGenerator.XamlFileGenerator.BuildBindingOption(XamlMemberDefinition m, INamedTypeSymbol propertyType) в C:\a\1\s\src\SourceGenerators\Uno.UI.SourceGenerators\XamlGenerator\XamlFileGenerator.cs:line 5023

The same happens for inline binding declaration like                     
`<Setter Property="Text" Value="{Binding Path=Whatever, Converter={StaticResource SomeConverter}}"/>`
with and without converter parameter.

**Comment by Youssef1313** [comment]
Possible fixed with https://github.com/unoplatform/uno/pull/16010 ? Not sure

**Comment by agneszitte** [comment]
> Possible fixed with #16010 ? Not sure

@MartinZikmund, maybe @morning4coffe-dev can help validate if it is already fixed or not

**Comment by morning4coffe-dev** [comment]
@MartinZikmund, @agneszitte, Tested on Skia, wasm, and Android with Uno `5.3.0-dev.1282` - all platforms now have the same behavior as WinUI for both scenarios. For the first one, I can see a red ‘Test’ text, and for the second one, although I see a blank page, it is the same behavior for me as on WinUI and it no longer results in any error.
